### ISSUE TITLE
Conflicting error on Android

### ISSUE THREAD

**Issue by dibelogrivaya** [square/react-native-square-reader-sdk#69]
Hi!

I get this error when I use Product Flavors
**INSTALL_FAILED_CONFLICTING_PROVIDER**

```
productFlavors {
        ver1 {
            flavorDimensions "environment"
            applicationIdSuffix ". ver1"
            versionNameSuffix "-ver1"
        }
        ver2 {
            flavorDimensions "environment"
            applicationIdSuffix ". ver2"
            versionNameSuffix "-ver2"
        }
    }

buildscript {
    ext {
        buildToolsVersion = "28.0.3"
        minSdkVersion = 21
        compileSdkVersion = 28
        targetSdkVersion = 27
        supportLibVersion = "28.0.0"
        sqipVersion = "1.1.0"
    }
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.3.1'
    }
}

```
<img width="511" alt="error" src="https://user-images.githubusercontent.com/1580342/62534708-656d4600-b852-11e9-9e0a-12cb758ec915.png">

**Comment by hukid** [comment]
@dibelogrivaya did you try to uninstall the app which was previously installed on your simulator or device?

**Comment by jedt** [comment]
We are also having this error when using productFlavors and when there are two apps on the android phone that uses the same library

`Failed to commit install session 1322935352 with command cmd package install-commit 1322935352. Error: INSTALL_FAILED_CONFLICTING_PROVIDER: Package couldn't be installed in /data/app/com.squarereadertest-vwp28GZkkEefRh0kcZLLKg==: Can't install because provider name com.squareup.sdk.reader.api.leak-sentry-installer (in package com.squarereadertest) is already used by [com.myfirstproject]`

**Comment by jedt** [comment]
I've made a reproducible demo project here https://github.com/jedt/SquareReaderTest there are instructions on the readme. Please let me know if I am missing something

**Comment by dibelogrivaya** [comment]
> @dibelogrivaya did you try to uninstall the app which was previously installed on your simulator or device?

Of course. I tried all possible scenarios.

**Comment by dibelogrivaya** [comment]
@hukid do you have any idea how to fix it?

**Comment by hukid** [comment]
@dibelogrivaya This is a bug in reader sdk, it uses hardcoded package name for the Leakcanary installer provider instead of picking application id. We started working on a way to substitute the package ID with placeholder.

We'll let you know if there is workaround.

**Comment by Armaxis** [comment]
Good news, there's a workaround for this issue you can use meanwhile.
The idea is to override the `provider`'s `android:authorities` property in **each** project with unique value, so they no longer conflict when apps are installed together. You can use project's `package name` for that; in example below I will assume that there are two projects with package names `com.squarereadertest` and `com.myfirstproject`

In the `AndroidManifest.xml` or first project add following inside the `<application>`:

```xml
<application 
   ...
    <provider
        android:name="shadow.leakcanary.internal.LeakSentryInstaller"
        android:exported="false"
        android:authorities="com.squarereadertest.leak-sentry-installer"
        tools:replace="android:authorities" />
    ...
</application>
```

In the `AndroidManifest.xml` or second project add following inside the `<application>`:

```xml
<application 
   ...
    <provider
        android:name="shadow.leakcanary.internal.LeakSentryInstaller"
        android:exported="false"
        android:authorities="com.myfirstproject.leak-sentry-installer"
        tools:replace="android:authorities" />
    ...
</application>
```

Please let us know if this works for you!

**Comment by dibelogrivaya** [comment]
@Armaxis thanks, it helped me!

**Comment by StephenJosey** [comment]
Hey all - this is fixed in the Reader SDK 1.3.5 release, so you do not need to use the workaround above, please just update. Closing this issue, but feel free to comment if you still encounter it or have other concerns.

**Comment by dibelogrivaya** [comment]
@StephenJosey all works fine, thanks!
